wrapanapi/systems/redfish.py

```diff
@@ -33,5 +71,80 @@ def _identifying_attrs(self):
     def info(self):
         return 'RedfishSystem url={}'.format(self.url)
 
-    def __del__(self):
-        """Disconnect from the API when the object is deleted"""
+    def server_stats(self, *requested_stats, **kwargs):
+        # Get the requester which represents the class of this method's caller
+        requester = kwargs.get('requester')
+
+        # Retrieve and return the stats
+        requested_stats = requested_stats or self._stats_available
+
+        return {stat: self._server_stats_available[stat](self, requester)
+                for stat in requested_stats}
+
+    def server_inventory(self, *requested_items, **kwargs):
+        # Get the requester which represents the class of this method's caller
+        requester = kwargs.get('requester')
+
+        # Retrieve and return the inventory
+        requested_items = requested_items or self._server_inventory_available
+
+        return {item: self._server_inventory_available[item](self, requester)
+                for item in requested_items}
+
+    def get_server(self, resource_id):
+        """
+        Fetch a Redfish resource that represents a particular physical server.
+
+        Args:
+          resource_id: the Redfish @odata.id of the resource representing the
+             server to be retrieved
+        """
+        return self.api_client.find(resource_id)
+
+    def num_servers(self):
+        return len(self.api_client.Systems.Members)
+
+    def get_server_cores(self, resource_id):
+        physical_system = self.get_server(resource_id)
+
+        return sum([int(p.TotalCores) for p in physical_system.Processors.Members])
+
+    def get_server_memory(self, resource_id):
+        """
+        Return the amount of memory, in MiB, for the server of the given
+        resource_id
+        """
+        return self.get_server(resource_id).MemorySummary.TotalSystemMemoryGiB * 1024
+
+    def get_num_server_firmwares(self, resource_id):
+        # not currently supported
+        return 0
+
+    def get_num_network_devices(self, resource_id):
+        # not currently supported
+        return 0
+
+    def get_num_storage_devices(self, resource_id):
+        # not currently supported
+        return 0
+
+    def get_server_power_status(self, resource_id):
+        """
+        Retrieve the current power status of the physical server.
+
+        Args:
+          resource_id: the Redfish @odata.id of the resource representing the
+             server to be retrieved
+        """
+        return self.get_server(resource_id).PowerState
+
+    def get_server_power_status_translated(self, resource_id):
```

Comment:
This method shouldn't exist here in wrapanapi, this has to do with MIQ specifically. VmState enumerations are what should be used to standardize mgmt class states within MIQ's integration_tests.

Reply:
👍